src/core/lib/event_engine/ares_resolver.cc

```diff
@@ -455,6 +458,9 @@ void AresResolver::LookupTXT(
 }
 
 void AresResolver::CheckSocketsLocked() {
+  if (channel_ == nullptr) {
```

Comment:
I see a lot of spots in this code that do not check if `channel_` is null. This is a new possible state with fork support changes, `channel_` always used to be valid for the life of the resolver. `channel_` is used 3x in `LookupHostname` without any null checks, for example. What guarantees that `channel_` is non-null here? Consider adding `CHECK` statements wherever `channel_` usage would be UB if it were null ... an explicit crash is usually better than memory corruption.

Reply:
Bifurcated resolver fixes and submitted separately.